packages/datadog-plugin-bullmq/test/index.spec.js

```diff
@@ -0,0 +1,351 @@
+'use strict'
+
+// IMPORTANT: Set DD_DATA_STREAMS_ENABLED BEFORE any requires
+process.env.DD_DATA_STREAMS_ENABLED = 'true'
+
+const { expect } = require('chai')
+const sinon = require('sinon')
+const { createIntegrationTestSuite } = require('../../dd-trace/test/setup/helpers/integration-test-helpers')
+const TestSetup = require('./test-setup')
+const DataStreamsContext = require('../../dd-trace/src/datastreams/context')
+const { computePathwayHash } = require('../../dd-trace/src/datastreams/pathway')
+const { ENTRY_PARENT_HASH, DataStreamsProcessor } = require('../../dd-trace/src/datastreams/processor')
+
+const testSetup = new TestSetup()
+
+const getDsmPathwayHash = (queueName, isProducer, parentHash) => {
+  const edgeTags = isProducer
+    ? ['direction:out', `topic:${queueName}`, 'type:bullmq']
+    : ['direction:in', `topic:${queueName}`, 'type:bullmq']
+  edgeTags.sort()
+  return computePathwayHash('test', 'tester', edgeTags, parentHash)
+}
+
+createIntegrationTestSuite('bullmq', 'bullmq', testSetup, {
+  category: 'messaging'
+}, (meta) => {
+  const { agent } = meta
+
+  describe('Queue.add() - bullmq.add', () => {
+    it('should generate span with correct tags (happy path)', async () => {
+      const traceAssertion = agent.assertSomeTraces((traces) => {
+        const spans = traces[0]
```

Comment:
Would prefer using the assertFirstTrace function that supports including the span object shape instead of separate assertion lines. assertFirstTrace name may be confusing the agent.